src/MobileUI/Services/AuthenticationService.cs

```diff
@@ -42,6 +44,49 @@ public AuthenticationService(IBrowser browser)
         };
     }
 
+    public async Task AutologinAsync(string accessToken)
+    {
+        try
+        {
+            var tokenHandler = new JwtSecurityTokenHandler();
+            var jwtToken = tokenHandler.ReadJwtToken(accessToken);
+
+            _accessToken = accessToken;
+            _tokenExpiry = jwtToken.ValidTo;
+
+            try
+            {
+                Preferences.Set(nameof(HasCachedAccount), true);
+                DetailsUpdated?.Invoke(this, EventArgs.Empty);
+
+                await MainThread.InvokeOnMainThreadAsync(async () =>
+                {
+                    Application.Current.MainPage = new AppShell();
+
+                    await Shell.Current.GoToAsync("//main");
+                });
```

Comment:
I've seen this at least twice.
Consider doing a utility static method GetMMainAppShell(). Maybe as an extension method to Application.Current.